src/components/Waypoints/WaypointIcon.tsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+import SelectedWaypoint from "@app/assets/waypoints/waypoint_selected.svg";
+import DefaultWaypoint from "@app/assets/waypoints/waypoint_default.svg";
+
+// File contains the waypoint icon and interface for two booleans that control its appearance
+// Called in Waypoints.tsx, once for each node detected
+
+export interface IWaypointProps {
+  isSelected: boolean;
+  expired: boolean;
+}
+
+const WaypointIcon = ({ isSelected, expired }: IWaypointProps) => {
+  return (
+    <img
+      className={"drop-shadow-lg"}
+      src={expired ? "" : isSelected ? SelectedWaypoint : DefaultWaypoint}
```

Comment:
Why do we render the `WaypointIcon` at all if a waypoint is expired?

Reply:
Since we get the list of waypoints from each node echoing it back, the expired waypoints are still in the list. Since the waypoint.map in `Map.tsx` isn't refreshing frequently, this prevents it from being rendered

Reply:
Okay, might be worth putting a ticket for blocking the render of this component when the waypoint has been deleted but seems fine for now